### PR TITLE
codegen+runtime: Module resolution fixes/ basic formatters for Array/Set/Dictionary iterators

### PR DESCRIPTION
**Pull request by robryanx**

### CODE REVIEW

File: selfhost/typechecker.jakt [hunk 1]
```diff
@@ -2719,7 +2719,7 @@ struct Typechecker {
                     is_instantiated: false
                     parsed_function: None
                     is_comptime: false
-                    is_closure: false
+                    is_closure: true
```
Comment (alimpfard) [inline selfhost/typechecker.jakt:2722]: Isn't this a function expression without any captures? that's not what this flag is for.
Maybe replace this field to be an enum?

Comment (robryanx) [reply selfhost/typechecker.jakt]: Ah, I have updated it to add to `FunctionType`
`Expression` and `Closure`
Let me know if that is okay, it seems that there are no `is_closure=true` cases currently so I am not sure that it is fine for that case.

Comment (jntrnr) [reply selfhost/typechecker.jakt]: pinging @alimpfard - does the above fix the issue you saw?

Comment (alimpfard) [reply selfhost/typechecker.jakt]: Looks right, sorry I forgot about this one :sweat_smile:

**Comment by robryanx** [comment]
@jntrnr Just noticed this had conflicts, fixed now.

**Pull request merged**